src/core/dbus-service.c

```diff
@@ -718,6 +748,53 @@ static int bus_service_set_transient_property(
         if (streq(name, "ReloadSignal"))
                 return bus_set_transient_reload_signal(u, name, &s->reload_signal, message, flags, error);
 
+        if (streq(name, "ExtraFileDescriptors")) {
+                int fd;
+                const char *fdname;
+
+                r = sd_bus_message_enter_container(message, 'a', "(hs)");
+                if (r < 0)
+                        return r;
+
+                while ((r = sd_bus_message_read(message, "(hs)", &fd, &fdname)) > 0) {
+                        _cleanup_(extra_file_descriptor_freep) ExtraFileDescriptor *extra_fd = NULL;
+                        _cleanup_close_ int dup_fd = -EBADF;
+
+                        extra_fd = new(ExtraFileDescriptor, 1);
+                        if (!extra_fd)
+                                return -ENOMEM;
+
+                        dup_fd = fcntl(fd, F_DUPFD_CLOEXEC, 3);
+                        if (dup_fd < 0)
+                                return -errno;
+
+                        *extra_fd = (ExtraFileDescriptor) {
+                                .fd = TAKE_FD(dup_fd),
+                                .fdname = strdup(fdname),
+                        };
```

Comment:
please do this initialization right after the new() above, i.e. swap this with the fcntl() blick.

Otherwise if the error paths on the fcntl() is hit you'll destroy an uninitialized object.